src/providers/postgres/qgsconnectionconfigurator.h

```diff
@@ -0,0 +1,190 @@
+      }
+      else
+      {
+        uri.setConnection( host, port, database, username, password, sslmode, authcfg );
+      }
+      uri.setUseEstimatedMetadata( estimatedMetadata );
+
+      return uri;
+    }
+
+
+    static bool publicSchemaOnly( const QString &connectionName )
+    {
+      QgsSettings settings;
+      return settings.value( connection_name_prefix.arg( connectionName ) + QStringLiteral( "/publicOnly" ), false ).toBool();
+    }
+
+    static bool geometryColumnsOnly( const QString &connectionName )
+    {
+      QgsSettings settings;
+
+      return settings.value( connection_name_prefix.arg( connectionName ) + QStringLiteral( "/geometryColumnsOnly" ), false ).toBool();
+    }
+
+    static bool dontResolveType( const QString &connectionName )
+    {
+      QgsSettings settings;
+
+      return settings.value( connection_name_prefix.arg( connectionName ) + QStringLiteral( "/dontResolveType" ), false ).toBool();
+    }
+
+    static bool useEstimatedMetadata( const QString &connectionName )
+    {
+      QgsSettings settings;
+
+      return settings.value( connection_name_prefix.arg( connectionName ) + QStringLiteral( "/estimatedMetadata" ), false ).toBool();
+    }
+
+    static bool allowGeometrylessTables( const QString &connectionName )
+    {
+      QgsSettings settings;
+      return settings.value( connection_name_prefix.arg( connectionName ) + QStringLiteral( "/allowGeometrylessTables" ), false ).toBool();
+    }
+
+    static bool allowProjectsInDatabase( const QString &connectionName )
+    {
+      QgsSettings settings;
+      return settings.value( connection_name_prefix.arg( connectionName ) + QStringLiteral( "/projectsInDatabase" ), false ).toBool();
+    }
+
+    static void deleteConnection( const QString &connectionName )
+    {
+      QgsSettings settings;
+
+      QString key = connection_name_prefix .arg( connectionName );
+      settings.remove( key + QLatin1String( "/service" ) );
+      settings.remove( key + QLatin1String( "/host" ) );
+      settings.remove( key + QLatin1String( "/port" ) );
+      settings.remove( key + QLatin1String( "/database" ) );
+      settings.remove( key + QLatin1String( "/username" ) );
+      settings.remove( key + QLatin1String( "/password" ) );
+      settings.remove( key + QLatin1String( "/sslmode" ) );
+      settings.remove( key + QLatin1String( "/publicOnly" ) );
+      settings.remove( key + QLatin1String( "/geometryColumnsOnly" ) );
+      settings.remove( key + QLatin1String( "/allowGeometrylessTables" ) );
+      settings.remove( key + QLatin1String( "/estimatedMetadata" ) );
+      settings.remove( key + QLatin1String( "/saveUsername" ) );
+      settings.remove( key + QLatin1String( "/savePassword" ) );
+      settings.remove( key + QLatin1String( "/save" ) );
+      settings.remove( key + QLatin1String( "/authcfg" ) );
+      settings.remove( key + QLatin1String( "/keys" ) );
+      settings.remove( key );
```

Comment:
I 'm not sure: can't you just remove key to remove all its children?

Reply:
this is properly handled when using the new API